test/models/Reads/PensionSchemeAdministratorReadsSpec.scala

```diff
@@ -0,0 +1,129 @@
+      val input = Json.obj(
+        "legalStatus" -> "Individual",
+        "sapNumber" -> "NumberTest",
+        "noIdentifier" -> JsBoolean(true),
+        "customerType" -> "TestCustomer",
+        "contactDetails" -> Json.obj("phone" -> "07592113", "email" -> "test@test.com"),
+        "companyAddressYears" -> JsString("over_a_year"),
+        "companyAddressId" -> JsObject(Map("addressLine1" -> JsString("line1"),"addressLine2" -> JsString("line2"),"addressLine3" -> JsString("line3"),"addressLine4" -> JsString("line4"),"postalCode" -> JsString("NE1"),"countryCode" -> JsString("GB"))))
+
+      "We have a valid legalStatus" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.legalStatus mustEqual pensionSchemeAdministratorSample.legalStatus
+      }
+
+      "We have a valid sapNumber" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.sapNumber mustEqual pensionSchemeAdministratorSample.sapNumber
+      }
+
+      "We have a valid noIdentifier" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.noIdentifier mustEqual pensionSchemeAdministratorSample.noIdentifier
+      }
+
+      "We have valid customerType" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.customerType mustEqual pensionSchemeAdministratorSample.customerType
+      }
+
+      "We have a valid idType" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input + ("idType" -> JsString("TestId")))(apiReads).asOpt.value
+
+        result.idType mustEqual pensionSchemeAdministratorSample.idType
+      }
+
+      "We have a valid idNumber" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input + ("idNumber" -> JsString("TestIdNumber")))(apiReads).asOpt.value
+
+        result.idNumber mustEqual pensionSchemeAdministratorSample.idNumber
+      }
+
+      "We have a moreThanTenDirectors flag" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input + ("moreThanTenDirectors" -> JsBoolean(true)))(apiReads).asOpt.value
+
+        result.numberOfDirectorOrPartners.value.isMorethanTenDirectors mustEqual pensionSchemeAdministratorSample.numberOfDirectorOrPartners.value.isMorethanTenDirectors
+      }
+
+      "We don't have moreThanTenDirectors flag" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.numberOfDirectorOrPartners mustEqual None
+      }
+
+      "We have contact details" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.correspondenceContactDetail.telephone mustBe pensionSchemeAdministratorSample.correspondenceContactDetail.telephone
+      }
+
+      "We have previous address details" in {
```

Comment:
Is there a previous address in the input JSON?

Reply:
No because it is over a year (not needed)...Here we are testing that the actual flag is correct.